checker.c

```diff
@@ -1,21 +1,77 @@
 #include <stdio.h>
 #include <assert.h>
+#include "checker.h"
+
+int compareOneLimit (float value, float limit){
+    int retval=1;
+    if(value > limit) {
+        printf("out of range!\n");
+        retval = 0;
+    }
+    else {
+        printf("OK!\n");
+    }
+    return retval;
+}
+
+int compareTwoLimits (float value, float minlimit, float uplimit){
+    int retval=1;
+    if(value < minlimit || value > uplimit) {
+        printf("out of range!\n");
+        retval = 0;
+    }
+    else {
+        printf("OK!\n");
+    }
+    return retval;
+}
+
+int checkTemperature(float temperature) {
+    printf("Temperature ");
+    return compareTwoLimits (temperature, MINTEMP, MAXTEMP);
+}
+int checkSoc(float soc) {
+    printf("SOC ");
+    return compareTwoLimits (soc, MINSOC, MAXSOC);
+}
+int checkChargeRate(float chargeRate){
+    printf("Charge Rate ");
+    return compareOneLimit (chargeRate, MINCHARGE);
+}
 
 int batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    printf("Temperature out of range!\n");
-    return 0;
-  } else if(soc < 20 || soc > 80) {
-    printf("State of Charge out of range!\n");
-    return 0;
-  } else if(chargeRate > 0.8) {
-    printf("Charge Rate out of range!\n");
-    return 0;
-  }
-  return 1;
+    return ((checkTemperature(temperature))&&(checkSoc(soc))&&(checkChargeRate(chargeRate)));
 }
 
 int main() {
```

Comment:
the range check functions maybe moved to a file specific for this content and the test asserts can be separated from production code.
Right now everything is in one file and that adds to complexity of code from a readability and maintainability perspective.